org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceExtension.java

```diff
@@ -60,16 +63,12 @@
  * }
  * </pre>
  */
-public class ServiceExtension implements BeforeEachCallback, ParameterResolver {
-
-	final static Namespace NAMESPACE = Namespace.create(ServiceExtension.class);
+public class ServiceExtension implements BeforeAllCallback, BeforeEachCallback, ParameterResolver {
 
 	@Override
-	public void beforeEach(ExtensionContext extensionContext) throws Exception {
-		List<Field> fields = findAnnotatedFields(extensionContext.getRequiredTestClass(), InjectService.class);
-
-		BundleContext bundleContext = FrameworkUtil.getBundle(extensionContext.getRequiredTestClass())
-			.getBundleContext();
+	public void beforeAll(ExtensionContext extensionContext) throws Exception {
```

Comment:
Note that for completeness, this should also inject into non-static fields if the `ExtensionContext` has a TestInstance present. This is for if you are running your tests with the `@TestInstance(PER_CLASS)`. There are a couple of situations where this is useful (as I have recently discovered) - if you're using Kotlin (which doesn't support `static`), or if you're using a `@Nested` test (as non-static inner classes cannot have static methods).

`BundleContextExtension` doesn't support this yet either, so unless you're feeling particularly keen then maybe leave this for now and we'll look at a future PR. :smile:

Reply:
We should fix both in a new PR!